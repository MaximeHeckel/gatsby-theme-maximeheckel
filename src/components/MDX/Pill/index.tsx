import React from 'react';
import styled from '@emotion/styled';
import { useTheme } from '../../../context/ThemeContext';

export enum PillVariant {
  INFO = 'info',
  SUCCESS = 'success',
  WARNING = 'warning',
  DANGER = 'danger',
}

interface PillProps {
  variant: PillVariant;
}

const PillWrapper = styled('span')<PillProps & { dark: boolean }>`
  font-size: 14px;
  border-radius: var(--border-radius-1);
  height: 28px;
  display: inline-flex !important;
  align-items: center;
  justify-content: center;
  padding: 5px 8px 5px 8px !important;
  min-width: 40px;
  font-weight: 500;
  cursor: default;
  user-select: none;
  ${(p) =>
    p.dark
      ? `
    ${
      p.variant === PillVariant.INFO
        ? `
          background: var(--maximeheckel-colors-emphasis);
          color: var(--maximeheckel-colors-brand);`
        : ''
    }
  
    ${
      p.variant === PillVariant.SUCCESS
        ? `
          background: var(--maximeheckel-colors-success-emphasis);
          color: var(--maximeheckel-colors-success);`
        : ''
    }
  
    ${
      p.variant === PillVariant.WARNING
        ? `
          background: var(--maximeheckel-colors-warning-emphasis);
          color: var(--maximeheckel-colors-warning);`
        : ''
    }
  
    ${
      p.variant === PillVariant.DANGER
        ? `
          background: var(--maximeheckel-colors-danger-emphasis);
          color: var(--maximeheckel-colors-danger);`
        : ''
    }
    `
      : `
      ${
        p.variant === PillVariant.INFO
          ? `
            background: var(--maximeheckel-colors-emphasis);
            color: var(--maximeheckel-colors-brand);`
          : ''
      }
    
      ${
        p.variant === PillVariant.SUCCESS
          ? `
            background: var(--maximeheckel-colors-success-emphasis);
            color: hsl(var(--palette-green-80));`
          : ''
      }
    
      ${
        p.variant === PillVariant.WARNING
          ? `
            background: var(--maximeheckel-colors-warning-emphasis);
            color: hsl(var(--palette-orange-80));`
          : ''
      }
    
      ${
        p.variant === PillVariant.DANGER
          ? `
            background: var(--maximeheckel-colors-danger-emphasis);
            color: var(--maximeheckel-colors-danger);`
          : ''
      }
      `}
`;

const Pill: React.FC<PillProps> = (props) => {
  const theme = useTheme();
  const { children, variant } = props;
  return (
    <PillWrapper {...props} dark={theme.dark} variant={variant}>
      {children}
    </PillWrapper>
  );
};

export default Pill;
