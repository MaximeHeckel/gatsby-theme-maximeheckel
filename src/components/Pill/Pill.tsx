import React from 'react';
import styled from '@emotion/styled';

interface PillWrapperProps {
  color: string;
}

const PillWrapper = styled('span')<PillWrapperProps>`
  font-size: 14px;
  background: ${(p) => p.color};
  border-radius: var(--border-radius-1);
  height: 22px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  padding: 4px 8px 5px 8px;
  margin-left: 8px;
  color: #2b2d3e;
  min-width: 40px;
`;

interface Props extends PillWrapperProps {
  text: string;
}

const Pill: React.FC<Props> = (props) => (
  <PillWrapper color={props.color}>{props.text}</PillWrapper>
);

export { Pill };
