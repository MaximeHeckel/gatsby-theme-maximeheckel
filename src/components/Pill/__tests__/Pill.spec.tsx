import { cleanup, render } from '@testing-library/react';
import React from 'react';
import Pill from '../';

afterEach(() => {
  cleanup();
});

jest.mock('gatsby-theme-maximeheckel/src/utils/styled', () => {
  const styled = require('@emotion/styled');
  return styled;
});

describe('Pill', () => {
  it('Renders the pill component properly', () => {
    const { getByText, asFragment } = render(<Pill color="blue" text="test" />);
    expect(getByText('test')).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
