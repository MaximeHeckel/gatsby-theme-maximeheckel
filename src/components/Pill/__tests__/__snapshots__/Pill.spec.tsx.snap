// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pill Renders the pill component properly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 14px;
  background: blue;
  border-radius: 10px;
  height: 22px;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4px 8px 5px 8px;
  margin-left: 8px;
  color: #2b2d3e;
  min-width: 40px;
}

<span
    class="emotion-0"
    color="blue"
  >
    test
  </span>
</DocumentFragment>
`;
